### DISCUSSION TITLE
Setup recording length after last detected object

### DISCUSSION THREAD

**Discussion by TheSleepySlee** [blakeblackshear/frigate#3317] [Ask A Question]
Hello

Been using frigate for a long time now.
Switched on v0.10 since some weeks. Working fine.

One thing that I miss is about how to configure length to record after last detected motion. Because sometimes recording stops while object is sill there.
My record settings are set like below. Thanks for your help

record:
  enabled: True
  retain:
    days: 7
    mode: motion
  events:
    retain:
      default: 14
      mode: active_objects

**Comment by NickM-27** [comment]
It isn't 100% clear what you're looking for. 

If you are mainly referring to events, then you should be able to use the `post_capture` attribute of events to capture time after the event has ended.

Otherwise if you want general recordings to maintain this then need to change the `record -> mode` to `all`
For this to apply to events as well then that one should be `all` too

**Reply by TheSleepySlee** [reply]
thanks, post_catpure seems to be what I was looking for. tried to add pre/post capture and will mark as answered if works correctly.
could not find this info in documentation.

**Reply by TEfjLONjDON** [reply]
I know this is not my thread, and I don't want to completely hijack this discussion. However if I understood your post above correctly...

> It isn't 100% clear what you're looking for.
> 
> If you are mainly referring to events, then you should be able to use the `post_capture` attribute of events to capture time after the event has ended.
> 
> Otherwise if you want general recordings to maintain this then need to change the `record -> mode` to `all` For this to apply to events as well then that one should be `all` too

then 

If I set `record -> mode` to `all`. Then would that lengthen the default 24/7 base recording segments from the default 10 seconds to a different value? I'm satisfied with my event clips, and I setup my `post_capture` along with `pre_capture` and `max_seconds` when I originally deployed. 

**_What I'm hoping to accomplish is to lengthen the default record segment time to 60 second clips. This would give me 60 actual .mp4 video segments instead of 360,_** to search threw if events didn't capture what is was looking for for some reason. 

This is actually what happened yesterday. Some SOB's decided to drive down the street and stop in-front of several of the houses, get out of the car run up to the vehicles in the yard and try to open them checking for unlocked doors.

Frigate didn't even register an event for this for some reason...  Video [here](https://photos.app.goo.gl/j5qXZVrJccwfuQjV9) if your interested in seeing this in action for your self. 

My problem is because this didn't trigger an event ((some how)) by the time i got home from work and my wife told me about she could not tell me what time it happened, so I went through 3 hours of video clips 360 x 3 = 1080 clips looking for the incident. 

I'll include my current record config below...

```
record:
  enabled: True
  expire_interval: 60
  retain:
    days: 2
    mode: all
  events:
    max_seconds: 300
    pre_capture: 3
    post_capture: 3
    objects:
      - person
    required_zones: []
    retain:
      default: 2
      mode: all
      objects:
        person: 3
```

**Reply by NickM-27** [reply]
> If I set record -> mode to all. Then would that lengthen the default 24/7 base recording segments from the default 10 seconds to a different value?

No, all that does is control which 10 second segments are kept (in this case changing it from 10 second segments that contain motion to all)

> What I'm hoping to accomplish is to lengthen the default record segment time to 60 second clips. This would give me 60 actual .mp4 video segments instead of 360, to search threw if events didn't capture what is was looking for for some reason.

There is a way to do this is by setting the record output args

`record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an`

NOTE: I recall (will need to see if I can find it) that blakeblackshear said it is not recommended and could lead to some unwanted behavior. Also based on your reason for changing this I am not sure if this is a good solution

> My problem is because this didn't trigger an event ((some how))

You can use your recorded footage to input as a camera in frigate to watch the debug view and see what it saw (make sure the settings are applied same as the actual camera): https://docs.frigate.video/contributing#2-create-a-local-config-file-for-testing

> My problem is because this didn't trigger an event ((some how)) by the time i got home from work and my wife told me about she could not tell me what time it happened, so I went through 3 hours of video clips 360 x 3 = 1080 clips looking for the incident.

Why did you go through the individual clips as opposed to just using the built-in recording viewing UI?

**Reply by TEfjLONjDON** [reply]
I went through the clips because those 2 camera out by the street are both being recorded in .h265 due to the size of the ssd used by home-assistant. The Frigate interface is not viewable for those two cameras even from edge browser.

```
record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
```

ill experiment with this, thank you so much for mentioning this code, my guess is that it make the database slower because of the size of the clip and the limited amount of cache that hassio offers to frigate `shim_size`

**Reply by NickM-27** [reply]
It will lead to a lot more storage used since if motion occurs in 1 of the 60 seconds, the entire 60 second segment will be retained. Before only 10 second segment would be retained
